src/crimson/os/seastore/extent_placement_manager.cc

```diff
@@ -790,19 +792,38 @@ RandomBlockOolWriter::do_write(
     stats.num_records += 1;
 
     ex->prepare_write();
-    return rbm->write(paddr,
-      ex->get_bptr()
+    uint64_t offset = 0;
+    bufferptr bp = ex->get_bptr();
+    if (is_inplace_rewrite(t, ex)) {
+    std::optional<std::pair<uint64_t, extent_len_t>> r;
+      if (ex->get_type() == extent_types_t::OBJECT_DATA_BLOCK) {
+	ObjectDataBlock *p = static_cast<ObjectDataBlock*>(ex.get());
+	r = p->get_modified_region();
+      } else {
+	TestBlock *p = static_cast<TestBlock*>(ex.get());
+	r = p->get_modified_region();
+      }
+      // r may be empty in case the extent has been replayed recently
+      // ---the extent would have no delta in this case
```

Comment:
IIUC this should be impossible to happen.

Reply:
After replay is done, all deltas are applied to the cache's buffer without adding deltas, right?
In this case, we need to overwrite the whole extent to the RBM even though there is no delta entries.

Reply:
Since we can not keep track of the modified region using deltas (1. after replay, 2. duplicate_for_write), I introduced `modified_region `in DATA_BLOCK. Please see the updated change.

Reply:
> After replay is done, all deltas are applied to the cache's buffer without adding deltas, right?
In this case, we need to overwrite the whole extent to the RBM even though there is no delta entries.

Hmm, in this case, after replaying, if the loaded deltas are kept with the dirty extents, the write here can still be bounded within the loaded delta range, right?

And the dirty extent would still have deltas in this case.

Reply:
Right, but, do we really need to maintain deltas here?--- Deltas unnecessarily occupy memory space, which is why I introduced `modified_range `in both `OBJECT_DATA_BLOCK `and `TEST_BLOCK`.

Moreover, if two mutations are called in a row to modify an exclusive range respectively, we probably can not track the former range because the delta is not copied to the new extent, right? To solve this, a clear way is to add the `modified_range`, like:
```
  TestBlock(const TestBlock &other)
    : LogicalCachedExtent(other) {}
    : LogicalCachedExtent(other), modified_region(other.modified_region) {}
```
Therefore, I think the `modified_range ` is necessary somehow.
What do you think?

Reply:
Make sense to me. IIUC inplace-rewrite delta payload is only needed until the mutated block is committed -- we should reset them upon commited/replayed, and keep track of the modified range instead.

In terms of the modified range, I think a similar statement is still true that "For the dirty extents that can do inplace-rewrite, the modified range must be valid upon rewrite", right?

And we only reset the modified range upon committed.

Reply:
Right. I addressed your comment. Please see the updated commit.